driver/src/main/java/org/neo4j/driver/v1/FieldAccess.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver.v1;
+
+/**
+ * Access to the interior fields of an underlying collection like
+ * data structure
+ */
+public interface FieldAccess
```

Comment:
Zhen: I like that API - but I agree with Stefan that `View` might end up
being confusing to use. I can't really think of anything better than `Like`
or `Access` as an alternative.

Stefan: Aren't they just `Values`? A collection of Values?



On Tue, Dec 1, 2015 at 3:46 AM, Zhen Li <notifications@github.com> wrote:

> In driver/src/main/java/org/neo4j/driver/v1/FieldAccess.java
> <https://github.com/neo4j/neo4j-java-driver/pull/86#discussion_r46257763>:
>
> > + *
> > + *     http://www.apache.org/licenses/LICENSE-2.0
> > + *
> > + * Unless required by applicable law or agreed to in writing, software
> > + * distributed under the License is distributed on an "AS IS" BASIS,
> > + * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > + * See the License for the specific language governing permissions and
> > + * limitations under the License.
> > + */
> > +package org.neo4j.driver.v1;
> > +
> > +/**
> > + * Access to the interior fields of an underlying collection like
> > + * data structure
> > + */
> > +public interface FieldAccess
>
> How about sth like:
>
> interface CollectionView
> {
>   int size();
>   boolean isEmpty();
>   Iterable<Value> values();
>   <T> Iterable<T> values( Function<Value, T> mapFunction );
> }
>
> Then for any XXView that might introduced in Cypher, we call them
> XXCypherView in the future?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/neo4j/neo4j-java-driver/pull/86/files#r46257763>.
>